### PR TITLE
Baystation12 merchant port

### PR DESCRIPTION
**Pull request by Alberyk**
Port of the merchant mechanics from baystation12.
All of the trader, prices and most essential system are done, as well the job datum and the like. Mapping the shuttle, the base and a new dock will be done in a later pr.
Also adds phazon construction and several other items used in this pr.
The chance of the merchant slow being open at round start can be changed via the config, it is 20% by default.

### CODE REVIEW

File: code/modules/modular_computers/file_system/programs/app_presets.dm [hunk 1]
```diff
@@ -162,3 +162,18 @@
...
+	description = "A preset for the merchant console"
+	available = 0
+/datum/modular_computer_app_presets/merchant/return_install_programs()
+	var/list/_prg_list = list(
+		new/datum/computer_file/program/filemanager(),
+		new/datum/computer_file/program/chatclient(),
+		new /datum/computer_file/program/merchant()
+	)
+	return _prg_list
+	return _prg_list
```
Comment (Arrow768) [inline code/modules/modular_computers/file_system/programs/app_presets.dm]: Double return

**Comment by skull132** [comment]
How would this affect your work, @Arrow768 ?

**Comment by Arrow768** [comment]
As discussed in discord. Not really an issue as long as the merchant is rare enough

File: code/datums/trading/trade.dm [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+	if(possible_origins && possible_origins.len)
+		origin = pick(possible_origins)
+
+	for(var/i in 3 to 6)
+		add_to_pool(trading_items, possible_trading_items, force = 1)
+		add_to_pool(wanted_items, possible_wanted_items, force = 1)
+
+//If this hits 0 then they decide to up and leave.
+/datum/trader/proc/tick()
+	spawn(0)
```
Comment (skull132) [inline code/datums/trading/trade.dm]: `INVOKE_ASYNC(proc_name)`

File: code/datums/trading/weaponry.dm [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+	possible_trading_items = list(
+		/obj/item/weapon/gun/projectile/pistol               = TRADER_ALL,
+		/obj/item/weapon/gun/projectile/colt                 = TRADER_ALL,
+		/obj/item/weapon/gun/projectile/sec                  = TRADER_ALL,
+		/obj/item/weapon/gun/projectile/shotgun/pump         = TRADER_ALL,
+		/obj/item/weapon/gun/projectile/shotgun/doublebarrel = TRADER_ALL,
+		/obj/item/weapon/gun/projectile/tanto                = TRADER_THIS_TYPE,
+		/obj/item/weapon/gun/projectile/revolver/detective   = TRADER_THIS_TYPE,
+		/obj/item/weapon/gun/projectile/revolver/deckard     = TRADER_THIS_TYPE,
+		/obj/item/ammo_magazine/c45m                         = TRADER_ALL,,
```
Comment (skull132) [inline code/datums/trading/weaponry.dm]: Null entry in this list. Remove the second `,` at the end.

File: code/game/mecha/mecha_construction_paths.dm [hunk 4]
```diff
@@ -1069,17 +1069,319 @@
...
+/datum/construction/mecha/phazon_chassis/action(atom/used_atom,mob/user as mob)
+	return check_all_steps(used_atom,user)
+
+/datum/construction/mecha/phazon_chassis/spawn_result()
+	var/obj/item/mecha_parts/chassis/const_holder = holder
+	const_holder.construct = new /datum/construction/reversible/mecha/phazon(const_holder)
+	const_holder.icon = 'icons/mecha/mech_construction.dmi'
+	const_holder.icon_state = "phazon0"
+	const_holder.density = TRUE
+	spawn()
```
Comment (skull132) [inline code/game/mecha/mecha_construction_paths.dm]: `QDEL_IN(src, 0)` instead.

File: code/datums/trading/trade.dm (hunk 2 above)
Comment (Lohikar) [inline code/datums/trading/trade.dm]: This one looks like it might be better suited for a 0-wait timer.

Comment (skull132) [inline code/datums/trading/trade.dm]: Then `add_timer(CALLBACK(src, .proc/procname), 0)`. Whichever works.

**Pull request merged**